ambry-api/src/main/java/com.github.ambry/router/CryptoService.java

```diff
@@ -69,8 +79,17 @@ default ByteBuf encrypt(ByteBuf toEncrypt, T key) throws GeneralSecurityExceptio
    * @throws {@link GeneralSecurityException} on any exception with decryption
    */
   default ByteBuf decrypt(ByteBuf toDecrypt, T key) throws GeneralSecurityException {
-    ByteBuffer decrypted = decrypt(toDecrypt.nioBuffer(), key);
-    return Unpooled.wrappedBuffer(decrypted);
+    if (toDecrypt.nioBufferCount() == 1) {
+      return Unpooled.wrappedBuffer(decrypt(toDecrypt.nioBuffer(), key));
+    } else {
+      ByteBuf temp = ByteBufAllocator.DEFAULT.heapBuffer(toDecrypt.readableBytes());
+      try {
+        temp.writeBytes(toDecrypt);
+        return Unpooled.wrappedBuffer(decrypt(temp.nioBuffer(), key));
```

Comment:
Is it possible to create common util function to handle the logic in these two default method?